plugins/modules/ndo_l3out_template.py

```diff
@@ -0,0 +1,828 @@
+    - The name of the L3 Domain.
+    type: str
+  target_dscp:
+    description:
+    - The DHCP Level of the L3Out.
+    type: str
+    choices:
+      - af11
+      - af12
+      - af13
+      - af21
+      - af22
+      - af23
+      - af31
+      - af32
+      - af33
+      - af41
+      - af42
+      - af43
+      - cs0
+      - cs1
+      - cs2
+      - cs3
+      - cs4
+      - cs5
+      - cs6
+      - cs7
+      - expedited_forwarding
+      - unspecified
+      - voice_admit
+  pim:
+    description:
+    - The protocol independent multicast (PIM) flag of the L3Out.
+    - By default, PIM is disabled. To enable the PIM, Layer 3 Multicast must be enabled on the O(vrf).
+    type: bool
+  inbound_route_map:
+    description:
+    - The name of the Route Map Policy for Route Control that needs to be associated with inbound route map when the O(routing_protocols) is "bgp".
+    type: str
+    aliases: [ import_route, inbound_route ]
+  outbound_route_map:
+    description:
+    - The name of the Route Map Policy for Route Control that needs to be associated with outbound route map when the O(routing_protocols) is "bgp".
+    type: str
+    aliases: [ export_route, outbound_route ]
+  routing_protocols:
+    description:
+    - The routing protocols of the L3Out.
+    type: list
+    elements: str
+    choices: [bgp, ospf]
+  ospf_area_config:
+    description:
+    - The OSPF area configuration of the L3Out.
+    aliases: [ ospf_config ]
+    type: dict
+    suboptions:
+      ospf_area_id:
+        description:
+        - The area id of the OSPF area.
+        type: str
+        aliases: [ area_id ]
+        required: true
+      ospf_area_type:
+        description:
+        - The area type of the OSPF area.
+        type: str
+        choices: [regular, stub, nssa]
+        aliases: [ area_type ]
+        required: true
+      ospf_cost:
+        description:
+        - The cost of the OSPF area.
+        type: int
+        aliases: [ cost ]
+        required: true
+      originate_summary_lsa:
+        description:
+        - This option is for OSPF NSSA or Stub area.
+        - When this option is disabled, not only Type 4 and 5, but also Type 3 LSAs are not sent into the NSSA or Stub area by the border leaf.
+        - Instead, the border leaf creates and sends a default route to the area.
+        - If there is no Type 3 LSA in this area in the first place, a default route is not created.
+        type: bool
+        aliases: [ originate_lsa ]
+      send_redistributed_lsas:
+        description:
+        - This option is for the OSPF NSSA (not-so-stubby area).
+        - When this option is disabled, the redistributed routes are not sent into this NSSA area from the border leaf.
+        - This is typically used when the O(ospf_area_config.originate_summary_lsa) option is also disabled.
+        - Because disabling the O(ospf_area_config.originate_summary_lsa) option creates and sends a default route to the NSSA or stub area.
```

Comment:
```suggestion
        - Because disabling the O(ospf_area_config.originate_summary_lsa) option creates and sends a default route to the NSSA or Stub area.
```